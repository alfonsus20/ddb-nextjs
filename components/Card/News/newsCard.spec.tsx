import { render } from "@testing-library/react";
import { NewsCard } from "..";
import "jest-canvas-mock";

const mockedProps = {
  title: "Judul",
  content: "Konten",
  image:
    "https://gdyzsghhuucelkwpprwa.supabase.co/storage/v1/object/public/images/users/Mask group (1).png",
  id: 1,
  blurHash: "U9HeB^R50{w|4T^kE0s9y?tR0KXTp]IosA9Z",
  date: "2022-05-22 17:00:00",
};

describe("News Card Component unit test", () => {
  it("should render correctly", () => {
    const container = render(<NewsCard {...mockedProps} />);
    expect(container).toMatchSnapshot();
  });

  it("should render human readable date format", () => {
    const { getByTestId } = render(<NewsCard {...mockedProps} />);
    const date = getByTestId("post-date");
    expect(date.textContent).toEqual("Minggu, 22 Mei 2022");
  });
});
