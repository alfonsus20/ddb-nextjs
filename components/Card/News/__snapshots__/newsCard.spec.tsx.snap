// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News Card Component unit test should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <a
        class="css-13rbvqk"
        href="/berita/1"
      >
        <div
          class="css-12al88"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
          >
            <img
              alt="berita"
              data-nimg="fill"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center; filter: blur(20px); background-size: cover; background-image: url(data:image/png;base64,00); background-position: center;"
            />
            <noscript />
          </span>
        </div>
        <div
          class="chakra-stack css-13k2ve6"
        >
          <p
            class="chakra-text css-g5hezf"
            data-testid="post-date"
          >
            Minggu, 22 Mei 2022
          </p>
          <p
            class="chakra-text css-1xgfdn"
          >
            Judul
          </p>
          <div
            class="css-1x8dgvm"
          >
            Konten
          </div>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="css-13rbvqk"
      href="/berita/1"
    >
      <div
        class="css-12al88"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="berita"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center; filter: blur(20px); background-size: cover; background-image: url(data:image/png;base64,00); background-position: center;"
          />
          <noscript />
        </span>
      </div>
      <div
        class="chakra-stack css-13k2ve6"
      >
        <p
          class="chakra-text css-g5hezf"
          data-testid="post-date"
        >
          Minggu, 22 Mei 2022
        </p>
        <p
          class="chakra-text css-1xgfdn"
        >
          Judul
        </p>
        <div
          class="css-1x8dgvm"
        >
          Konten
        </div>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
